miden/src/benches/merkle_path_rescue_prime.rs

```diff
@@ -0,0 +1,212 @@
+    vec![1, 10, 100, 1000]
+}
+
+impl Benchmark for Job {
+    const NAME: &'static str = "merkle_rescue_prime";
+    type Spec = u32;
+    type ComputeOut = Vec<u64>;
+    type ProofType = miden_prover::StarkProof;
+
+    fn job_size(spec: &Self::Spec) -> u32 {
+        *spec
+    }
+
+    fn output_size_bytes(output: &Self::ComputeOut, _proof: &Self::ProofType) -> u32 {
+        (output.len() * 8) as u32
+    }
+
+    fn proof_size_bytes(proof: &Self::ProofType) -> u32 {
+        proof.to_bytes().len() as u32
+    }
+    
+    fn new(num_iter: Self::Spec) -> Self {
+        let source = format!(
+            "  
+            begin
+            # verify a merkle path
+                repeat.{}
+      
+                    # stack = [d, i, R, ...]
+                    # duplicate the index and use as counter
+                    dup.1 push.1 add movdn.6 
+                    # stack = [d, i, R, (i+1) ...]
+        
+                    # verify merkle path for node i
+                    mtree_get
+                    # [V, R, (i+1) ...]
+
+                    # now we check if V is what we expect.
+                    # our leafs look like this: [0, 0, 0, counter]
+                    # get the word with the counter copied back
+                    dup.8 push.1 sub 
+                    #[i, V, R, (i+1),  ...]
+                    
+                    # then make it a Word
+                    padw drop
+                    #[0, 0, 0, i, V, R, (i+1),  ...]
+
+
+                    # if the two top words are equal, 
+                    # [0, 0, 0, i] == V,
+                    # we have the expected leaf
+                    eqw
+
+                    # fails if top stack element is 0
+                    inv 
+
+                    # if it did not fail, we continue with the next iteration
+                    drop dropw dropw
+                    #[R, (i+1),  ...]
+                    
+                    movup.4
+                    push.32
+                    #[32, (i+1), R,  ...]
+                end
+                drop drop
+            end",
+            num_iter
+        );
+
+        // Compiling the program
+        let assembler = Assembler::default();
+        let program = assembler
+            .compile(source.as_str())
+            .expect("Could not compile source");
+
+        // Default (96 bits of security)
+        let proof_options = ProofOptions::with_96_bit_security();
+
+        let program_outputs = ProgramOutputs::new(vec![], vec![]);
+
+        // We first create an array of keys as basis of our Sparse Merkle Tree.
+        // For depth 32 we need 32^2 = 1024 keys. We then transform every number into a Word by adding Felt::Zero  
+        let merkle_leafs_keys: Vec<u64> = (0..1024).collect();
+        let mut merkle_leafs: Vec<Word> = Vec::new();
+
+        for i in 0..merkle_leafs_keys.len() {
+            merkle_leafs.push([Felt::new(merkle_leafs_keys[i]), Felt::ZERO, Felt::ZERO, Felt::ZERO]);
+        }
```

Comment:
Ah indeed, it is 2^32, which is a way bigger number `4,294,967,295`. 

However, since we only have to check 1000 leaf nodes at the most at job_size = 1000, it worked as well. 

Let me update the comment

Reply:
Ok, now it makes sense.